src/components/chat/commented-documents.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from "react";
+import { useUIStore, useStores } from "../../hooks/use-stores";
+import { useFirestore } from "../../hooks/firestore-hooks";
+import { CurriculumDocument } from "../../lib/firestore-schema";
+import { getSectionTitle } from "../../models/curriculum/section";
+import { UserModelType } from "../../models/stores/user";
+import "./commented-documents.scss";
+
+interface IProps {
+  documentObj: CurriculumDocument,
+  user?: UserModelType
+  handleDocumentClick: () => void;
+}
+
+// Not sure this is the best way to do this.  The issue, I think
+// Is that the promises Firebase returns are not visible to TypeScript compiler
+// which can't see that the docs will eventually have the fields that match a CurriculumDocument
+interface PromisedCurriculumDocument extends CurriculumDocument {
+  id?: string,
+  title?: string,
+  numComments?: number,
+}
+
+export const CommentedDocuments: React.FC<IProps> = ({documentObj, user, handleDocumentClick}) => {
+  const [docsCommentedOn, setDocsCommentedOn] = useState<PromisedCurriculumDocument[]>();
+  const [db] = useFirestore();
+  const cDocsRef = db.collection("curriculum");
+  const { unit, problem } = documentObj ;
+  const cDocsInScopeRef = cDocsRef
+    .where("unit", "==", unit)
+    .where("problem", "==", problem)
+    .where("network","==", user?.network);
+  const ui = useUIStore();
+  const stores = useStores();
+
+  useEffect(() => {
+    const unsubscribeFromDocs = cDocsInScopeRef.onSnapshot(async querySnapshot => {
+      const docs = querySnapshot.docs.map(doc => {
+        return (
+          {
+            id: doc.id,
+            title: "temp",
+            numComments: 0,
+            ...doc.data()
+          }
+        );
+      });
+      const commentedDocs: PromisedCurriculumDocument[] = [];
+
+      for (let doc of docs){
+        const docCommentsRef = cDocsRef.doc(doc.id).collection("comments");
+        await docCommentsRef.get().then((qs) => {
+          if (qs.empty === false) {
+            const firstCharPosition = doc.id.split("_", 4).join("_").length + 1; //first char after 4th _
+            const sectionType =  doc.id.substring(firstCharPosition, doc.id.length);
```

Comment:
You should really figure this out from the content of the doc not its id. The doc's content should list the section type as a property.

But it is fine to leave that fix for a later PR.

Reply:
If I console log`doc` above this, I see that it has a "section" key but Suprisingly, I can't access the string if I do` doc.section`. I can show you tomorrow, but I'll leave it as is now.